activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -27,6 +29,8 @@ def query(sql, name = nil) #:nodoc:
         private_constant :READ_QUERY
 
         def write_query?(sql) # :nodoc:
+          return false if sql.blank? || sql == ";"
```

Comment:
I am guessing you are going to want this in all the adapters not just pg,

Also what do you think of something like this?

```suggestion
          cleaned_query = sql.gsub(%r{\/\*(.*)\*\/}, "").strip # remove comments
          return false if cleaned_query.blank? || cleaned_query == ";"
```

Not sure this catches all cases and there is probably a better way to remove comments from the query.

Reply:
I wrote it for all adapters originally, but mysql and sqlite don't like empty queries. See https://buildkite.com/rails/rails/builds/78094#65a639d9-3b09-44a0-8829-e781591d7ac6 https://buildkite.com/rails/rails/builds/78094#ec0c5677-5c91-4258-b9fb-362273bb02dc

Reply:
I don't mind that suggestion, I'm wary of the performance impact of all this but I may be overthinking it.